library/nist_kw.c

```diff
@@ -455,7 +467,7 @@ int mbedtls_nist_kw_unwrap( mbedtls_nist_kw_context *ctx,
 cleanup:
     if( ret != 0 )
     {
-        mbedtls_platform_zeroize( output, *out_len - 1);
+        mbedtls_platform_zeroize( output, *out_len - 1 );
```

Comment:
`*out_len - 1` can't be right: why would you not zeroize the last byte of output?

And in any case, as you noted in https://github.com/ARMmbed/mbedtls/pull/1658#discussion_r199696156 this is wrong because `*out_len` does not represent the number of bytes written. You need to wipe at least the number of bytes written.

Note that you can use `memset` here, you don't need `zeroize`, since this is an output of the function which the compiler can't optimize away.

Reply:
I can't recall why the `-1` part. I think it's residue from when it was a different value.